### ISSUE TITLE
Broken PTO timeout calculation?

### ISSUE THREAD

**Issue by stablebits** [h2o/quicly#150]
Shouldn't something like this be used? Otherwise, r->pto_count is never bigger than 1. See below the details of a test scenario.

```
diff --git a/include/quicly/loss.h b/include/quicly/loss.h
index a73e8e9..6ed3151 100644
--- a/include/quicly/loss.h
+++ b/include/quicly/loss.h
@@ -231,8 +231,7 @@ inline int quicly_loss_on_alarm(quicly_loss_t *r, uint64_t largest_sent, uint64_
         return quicly_loss_detect_loss(r, largest_acked, do_detect);
     }
     /* PTO */
-    if (r->pto_count == 0)
-        ++r->pto_count;
+    ++r->pto_count;
     *num_packets_to_send = 2;
     return 0;
 }
```
Related: "Replaces uses of TLP and RTO with PTO #134"

Test case:

1) Run server
```
./cli -c t/assets/server.crt -k t/assets/server.key 0.0.0.0 4433
```
2) Run client and interrupt it almost immediately (Ctrl-C)
```
./cli localhost 4433 -p /100000000.txt > /dev/null
```

What can be seen with current code:

Lots of data (supposedly retransmits) sent by the server side for about 30 seconds (idle_timeout). Inter-packet delta is ~50-70ms.

```
[...]
22:16:00.052057 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:00.052061 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:00.052064 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:00.977012 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:00.977017 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:00.977020 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:02.829341 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:02.829368 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:02.829386 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:06.533556 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:06.533582 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:06.533600 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:13.941470 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:13.941496 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:13.941515 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:28.742270 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:28.742297 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:28.742315 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
```

**Comment by stablebits** [comment]
With this fix, timeouts are growing exponentially as (supposedly) expected.

```
22:16:00.052057 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:00.052061 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:00.052064 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:00.977012 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:00.977017 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:00.977020 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:02.829341 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:02.829368 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:02.829386 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:06.533556 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:06.533582 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:06.533600 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:13.941470 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:13.941496 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:13.941515 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:28.742270 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
22:16:28.742297 IP 127.0.0.1 > 127.0.0.1: ICMP 127.0.0.1 udp port 56806 unreachable, length 556
22:16:28.742315 IP 127.0.0.1.4433 > 127.0.0.1.56806: UDP, length 1280
```

**Comment by janaiyengar** [comment]
@stablebits : Thank you for catching this issue -- it is currently being fixed in #148.

**Comment by janaiyengar** [comment]
Closed in error, I'll leave this issue open.
